src/app/modules/form/admin-form/admin-form.payments.service.ts

```diff
@@ -81,3 +85,54 @@ export const updatePayments = (
 export const getPaymentGuideLink = (): string => {
   return paymentConfig.guideLink
 }
+
+/**
+ * Update the payments of the given form
+ * @param formId the id of the form to update the end page for
+ * @param newStartPage the new start page object to replace the current one
+ * @returns ok(updated start page object) when update is successful
+ * @returns err(FormNotFoundError) if form cannot be found
+ * @returns err(PossibleDatabaseError) if start page update fails
+ * @returns err(InvalidPaymentAmountError) if payment amount exceeds MAX_PAYMENT_AMOUNT
+ */
+export const updatePaymentsProduct = (
+  formId: string,
+  newProducts: PaymentsProductUpdateDto,
+): ResultAsync<
+  IEncryptedFormDocument['payments_field'],
+  PossibleDatabaseError | FormNotFoundError | InvalidPaymentAmountError
+> => {
+  // TODO: should ensure if amounts_cents is within range for array
```

Comment:
// TODO: compare each product max price with max qty and block
  // ensure that product.amount_cents * product.max_qty <= MAX_PAYMENT_AMOUNT

Reply:
48a0de02a7a7b08e97bbe26c425c01d3ac5fed2f